src/lib.rs

```diff
@@ -1236,6 +1236,58 @@ pub trait Itertools : Iterator {
     {
         self.sorted_by(cmp)
     }
+
+    /// Return the minimum and maximum element of an iterator.
+    ///
+    /// For the minimum, the first minimal element is returned.  For the maximum,
+    /// the last maximal element wins.  This matches the behavior of the standard
+    /// `Iterator::min()` and `Iterator::max()` methods.
+    ///
+    /// This requires `clone()`-able elements since the returned minimum and
+    /// maximum might actually be the same item.
+    fn minmax(mut self) -> Option<(Self::Item, Self::Item)>
+        where Self: Sized, Self::Item: Ord, Self::Item: Clone
```

Comment:
So far I followed std min and max, which require `Ord`. This might be a more useful addition with `PartialOrd`, but I'm not sure what the behavior for NaN should be - three not too unreasonable ones come to mind:

a) skip all NaNs, return None for only-NaN iterator
b) skip all NaNs, but return NaN for only-NaN iterator
c) if there is a NaN, return it as min and max

I also guess that if we want this, we have to create a custom trait `OrdOrFloat`?